lua/sim/Projectile.lua

```diff
@@ -163,19 +169,41 @@ Projectile = ClassProjectile(ProjectileMethods) {
             sy = sy + offset
             sz = sz + offset
 
-            local dx = (Random() - 0.5) * fuzziness * sx
-            local dy = (Random() - 0.5) * fuzziness * sy
-            local dz = (Random() - 0.5) * fuzziness * sz
+            local dx = sx * (Random() - 0.5) * fuzziness
+            local dy = sy * (Random() - 0.5) * fuzziness + cy * 2
+            local dz = sz * (Random() - 0.5) * fuzziness + cz * 2
+            local dw
 
-            local target = {
-                px + dx * mch - dz * msh,
-                py + cy + 0.5 * sy + dy,
-                pz + dx * msh + dz * mch,
-            }
+            -- Rotate a vector by a quaternion: q * v * conjugate(q)
+            -- Supreme Commander quaternions use y,z,x,w!
+            local ty, tz, tx, tw = unpack(target:GetOrientation())
+
+            dw = -tx * dx - tz * dy - ty * dz
+            dx = tw * dx + tz * dz - ty * dy
+            dy = tw * dy + ty * dx - tx * dz
+            dz = tw * dz + tx * dy - tz * dx
+
+            tx, tz, ty = -tx, -tz, -ty
+
+            dx = dw * tx + dx * tw + dy * ty - dz * tz
+            dy = dw * tz + dy * tw + dz * tx - dx * ty
+            dz = dw * ty + dz * tw + dx * tz - dy * tx
```

Comment:
These changes would break the code because variables are assigned new values in the middle of the computation. That is why the original code only used 1 assignment to change all the values.
```suggestion
            -- compute the product in a single assignment to not have to use temporary variables.
            dw, dx, dy, dz = 
            -tx * dx - tz * dy - ty * dz,
             tw * dx + tz * dz - ty * dy,
             tw * dy + ty * dx - tx * dz,
             tw * dz + tx * dy - tz * dx

            tx, tz, ty = -tx, -tz, -ty
            
            -- compute the product in a single assignment to not have to use temporary variables.
            dx, dy, dz = 
            dw * tx + dx * tw + dy * ty - dz * tz,
            dw * tz + dy * tw + dz * tx - dx * ty,
            dw * ty + dz * tw + dx * tz - dy * tx
```

Reply:
Ah, my mistake. That is a bit confusing, haha